.github/workflows/build-snap.yml

```diff
@@ -0,0 +1,20 @@
+name: Snap Builder
+run-name: ${{ github.actor }} is building the ${{ github.repository }}
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+  workflow_dispatch:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: snapcore/action-build@v1
+        id: build
+      - uses: actions/upload-artifact@v3
+        with:
+          name: snap
```

Comment:
Use the actual snap name to include info such as architecture and version. For example: `chip-tool_v1.2.0.1+snap_amd64.snap.zip` instead of `snap.zip`